bundles/pubsub/pubsub_admin_tcp/README.md

```diff
@@ -0,0 +1,112 @@
+
+### IP Addresses 
+
+To use TCP, 1 IP address is needed:
+This is the IP address that is bound to an (ethernet) interface (from the publisher). 
+This IP address is defined with  the `PSA_IP` property.
+For example: `PSA_IP=192.168.1.0/24`.
+Note the example uses CIDR notation, the CIDR notation specifies the IP range that is used
+by the pubsub admin for searching the network interfaces.
+
+Note the port will be automatically assigned, when none is specified.
+A fixed port can be assigned for example with: `PSA_IP=192.168.1.0/24:34000`.
+
+
+### Discovery
+
+When a publisher wants to publish a topic a TopicSender and publisher endpoint is created by the Pubsub Topology Manager.
+When a subscriber wants to subscribe to topic a TopicReceiver and subscriber endpoint is created by the Pubsub Topology Manager.
+The endpoints are published by the PubSubDiscovery within its topic in ETCD (i.e. udp://192.168.1.20:40123).
+A subscriber, interested in the topic, is informed by the TopologyManager that there is a new endpoint. 
+The TopicReceiver at the subscriber side creates a listening socket based on this endpoint.
+Now a data-connection is created and data send by the publisher will be received by the subscriber.  
+
+### Static endpoint
+
+TCP pubsub admin also be used as a static endpoint that can be used to communicate with external
+TCP publish/subsribers that don't use Discovery.
+---
+With the `tcp.static.bind.url` property, the static UDP bind can be configured. 
+For TCP the publisher topic properties should configure the `udp.static.bind.url` property to configure TCP bind interface
+that accepts incomming connections.
+
+---
+With the `udp.static.connect.urls` property, the static TCP connections can be configured. (Note The urls are space separate)
+For TCP the subscriber topic properties should configure the `udp.static.connect.urls` property 
+to configure TCP connections that subsribe to the data.
+
+---
+Note a special/dedicated protocol service implemenation can be used to communcate with external publisher/subscribers 
+
+---
+
+## Passive Endpoints
+
+Each TCP pubsub publisher/subscriber can be comfigured as a passive endpoint. 
+A passive endpoint reuses an existing socket of a publisher/subscriber using properties.
+This can be used to support full duplex socket connections.
+This can be used for both discovery and static end points.
+
+The topic `tcp.passive.key` property defines the unique key to select the socket.
+The `tcp.passive.key` property needs to be defined both in the topic properties of the topic connections that will be shared.
+and the topic properties of the 'passive' topic that will re-use the connection.
+The `tcp.passive.configured` topic property will indicate that is topic is passive and will re-use the connections
+indicated with the `tcp.passive.key` property
+
+For example: `tcp.passive.key=udp://localhost:9500` and `tcp.passive.configured=true`
```

Comment:
why a url as key. I think anything can be a key and by giving the example a `udp://localhost:9500` this conveys this this key is used for socket connections.. but it just a "id" / "key".